src/tests/ftest/util/server_utils.py

```diff
@@ -290,3 +273,35 @@ def kill_server(hosts):
     for host in hosts:
         subprocess.call("ssh {0} \"{1}\"".format(host, '; '.join(kill_cmds)),
                         shell=True)
+
+def clustershell_execute(cmd, hosts, timeout=60):
+    """
+    Python cluster shell execute function with error check.
+    Args:
+        cmd: Command in string to run on cluster
+        hosts: Cluster nodes
+        timeout: Command timeout
+    """
+    local_task = task_self()
+    local_task.run(cmd=cmd, nodes=hosts, timeout=timeout)
+    for rc_code, _keys in local_task.iter_retcodes():
+        if rc_code is not 0:
+            for output, _node in local_task.iter_buffers():
+                raise ServerFailed("Cluster Command Failed\n{} : RC={} :{}"
```

Comment:
I would suggest not raising an exception on the first fail, but rather collecting a list of messages for each failed remote command and raising an event with all of the messages.

Reply:
Done

Reply:
done